### PR TITLE
dcrsqlite: test RetrieveAllPoolValAndSize and RetrieveBlockFeeInfo, tier 3

### PR DESCRIPTION
**Pull request by JFixby**
This is tier 3 of the #514 
Testing:
- `dcrsqlite/sqlite.go` `DB.RetrieveAllPoolValAndSize()`
- `dcrsqlite/sqlite.go` `DB.RetrieveBlockFeeInfo()`

### CODE REVIEW

File: db/dcrsqlite/dcrsqlite_test.go [hunk 1]
```diff
@@ -30,3 +32,42 @@ func InitTestDB(targetDBFile string) *DB {
...
+	ExpectEmptyArray(testedFunctionName, "chartsData.SizeF", chartsData.SizeF)
+	ExpectEmptyArray(testedFunctionName, "chartsData.ValueF", chartsData.ValueF)
+	ExpectEmptyArray(testedFunctionName, "chartsData.Unspent", chartsData.Unspent)
+	ExpectEmptyArray(testedFunctionName, "chartsData.Revoked", chartsData.Revoked)
+	ExpectEmptyArray(testedFunctionName, "chartsData.Height", chartsData.Height)
+	ExpectEmptyArray(testedFunctionName, "chartsData.Pooled", chartsData.Pooled)
+	ExpectEmptyArray(testedFunctionName, "chartsData.Solo", chartsData.Solo)
+}
+
+func ExpectEmptyArray(
```
Comment (chappjc) [inline db/dcrsqlite/dcrsqlite_test.go]: Why does this have to take an interface and use reflect / .Len when you could simply test with `len`?  e.g. 

```if len(chartsData.TimeStr) != 0```

File: db/dcrsqlite/dcrsqlite_test.go [hunk 2]
```diff
@@ -30,3 +32,42 @@ func InitTestDB(targetDBFile string) *DB {
...
+//  - all the returned arrays expected to be empty
+//  - chartsData is not null (nil)
+//
+// These checks are sorted according to the ChartsData struct fields order
+func checkChartsDataIsDefault(testedFunctionName string, chartsData *dbtypes.ChartsData) {
+	if chartsData == nil {
+		testutil.ReportTestFailed(
+			testedFunctionName + "() failed: value is nil")
+	}
+	// Array checks:
```
Comment (chappjc) [inline db/dcrsqlite/dcrsqlite_test.go]: I don't see the necessity of all these lines.  If you want to make sure that `chartsData` is initialized to the zero values, just do:

```go
var emptyChartsData dbtypes.ChartsData
if !reflect.DeepEqual(emptyChartsData, *chartsData) {
```

File: db/dcrsqlite/retrieveallpoolvalandsize_test.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+func testEmptyDBRetrieveAllPoolValAndSize(db *DB) {
+	result, err := db.RetrieveAllPoolValAndSize()
+	if err != nil {
+		testutil.ReportTestFailed(
+			"RetrieveAllPoolValAndSize() failed: default result expected:",
+			err)
+	}
+	checkChartsDataIsDefault("RetrieveAllPoolValAndSize", result)
```
Comment (chappjc) [inline db/dcrsqlite/retrieveallpoolvalandsize_test.go]: So instead of all that other code, you can just check that the pointer is non-nil, and the data is default with `DeepEqual` as I said.  Or you could use something like https://github.com/google/go-cmp

File: db/dcrsqlite/retrieveblockfeeinfo_test.go [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import (
+	"testing"
+
+	"github.com/decred/dcrdata/testutil"
+)
+
+func TestEmptyDBRetrieveBlockFeeInfo(t *testing.T) {
+	testutil.BindCurrentTestSetup(t)
+	db := InitTestDB(DBPathForTest())
+	testEmptyDBRetrieveBlockFeeInfo(db)
```
Comment (chappjc) [inline db/dcrsqlite/retrieveblockfeeinfo_test.go]: Is it necessary to have `InitTestDB` run repeatedly.  It would be much more concise to have lots of these "empty" tests just use the one empty db you init.  This is one reason I referred you to golang's established test harnesss setup: https://golang.org/pkg/testing/#hdr-Main.

File: db/dcrsqlite/dcrsqlite_test.go (hunk 1 above)
Comment (JFixby) [reply db/dcrsqlite/dcrsqlite_test.go]: All the arrays above has different types: `uint[]`, `float[]`, `string[]` and so on. Reflect helps here to avoid creating a dedicated function for each array type.

File: db/dcrsqlite/dcrsqlite_test.go (hunk 2 above)
Comment (JFixby) [reply db/dcrsqlite/dcrsqlite_test.go]: DeepEqual wouldn't provide detailed info on what went wrong (which field is incorrect).

File: db/dcrsqlite/retrieveblockfeeinfo_test.go (hunk 4 above)
Comment (JFixby) [reply db/dcrsqlite/retrieveblockfeeinfo_test.go]: Thank you.  I'll look into it.

Comment (JFixby) [reply db/dcrsqlite/retrieveblockfeeinfo_test.go]: I found the following solution to address the issue: 
```
var ReusableEmptyDB *DB = nil

// ObtainReusableEmptyDB returns a single reusable instance of an empty DB. The instance
// is created once during the first call. All the subsequent calls will return
// result cached in the ReusableEmptyDB variable above.
func ObtainReusableEmptyDB() *DB {
	if ReusableEmptyDB == nil {
		testName := "ReusableEmptyDB"
		testutil.ResetTempFolder(&testName)
		target := filepath.Join(testName, testutil.DefaultDBFileName)
		targetDBFile := testutil.FilePathInsideTempDir(target)
		ReusableEmptyDB = InitTestDB(targetDBFile)
	}
	return ReusableEmptyDB
}
```

File: db/dcrsqlite/dcrsqlite_test.go (hunk 1 above)
Comment (chappjc) [reply db/dcrsqlite/dcrsqlite_test.go]: As I mentioned in my other comment, you don't need a dedicated function at all. `len(uintArray)`, `len(floatArray)`, all fine.

File: db/dcrsqlite/retrieveblockfeeinfo_test.go [hunk 5]
```diff
@@ -0,0 +1,23 @@
+package dcrsqlite
+
+import (
+	"testing"
+
+	"github.com/decred/dcrdata/testutil"
+)
+
+func TestEmptyDBRetrieveBlockFeeInfo(t *testing.T) {
+	testutil.BindCurrentTestSetup(t)
```
[the remark this answers is not in the document]
Comment (JFixby) [reply db/dcrsqlite/retrieveblockfeeinfo_test.go]: I really did.

**Comment by chappjc** [comment]
Thank you.  The final request I have is to combine `retrieveallpoolvalandsize_test.go` and `retrieveblockfeeinfo_test.go`.  There's no compelling reason to have the tests in different files.  They  both involve dcrsqlite, the blocks table, and retrieval (even empty DB tests).  The bar for keeping tests in the same file is even lower than this, so please do merge.

**Comment by JFixby** [comment]
Should be good now.
